src/test/java/com/example/ShortnerURL/ShortnerUrlApplicationTests.java

```diff
@@ -1,13 +1,126 @@
+
+	private ApiService apiService;
+	private ShortLinkEntity shortLinkEntity;
+	private ShortLinkDto shortLinkDto;
+
+	@BeforeEach
+	void setup() {
+		apiService = new ApiService(shortLinkRepository);
+		shortLinkEntity = ShortLinkEntity.builder()
+				.id(1L)
+				.longLink("https://news.sky.com/")
+				.shortLinkCode("NjxQpL")
+				.build();
+		shortLinkDto = ShortLinkDto.builder()
+				.shortLinkCode("NjxQpL")
+				.shortLink(DOMAIN + "NjxQpL")
+				.longLink("https://news.sky.com/")
+				.build();
+
+	}
+
+	@Test
+	void shouldGetAllShortLinks() {
+		when(shortLinkRepository.findAll()).thenReturn(List.of(shortLinkEntity));
+		List<ShortLinkDto> result = apiService.getAllShortLinks();
+		assertEquals(1, result.size());
+		ShortLinkDto actualDto = result.get(0);
+		assertEquals(shortLinkDto.getShortLink(), actualDto.getShortLink());
+		assertEquals(shortLinkDto.getShortLinkCode(), actualDto.getShortLinkCode());
+		assertEquals(shortLinkDto.getLongLink(), actualDto.getLongLink());
+	}
+
+	@Test
+	void shouldGetShortLink() {
+		when(shortLinkRepository.findByShortLinkCode(shortLinkEntity.getShortLinkCode())).thenReturn(List.of(shortLinkEntity));
+		ShortLinkDto result = apiService.getShortLink(shortLinkEntity.getShortLinkCode());
+		assertNotNull(result);
+		assertEquals(shortLinkEntity.getShortLinkCode(), result.getShortLinkCode());
+		assertEquals(shortLinkEntity.getLongLink(), result.getLongLink());
+		assertEquals(DOMAIN + shortLinkEntity.getShortLinkCode(), result.getShortLink());
+		verify(shortLinkRepository, times(1)).findByShortLinkCode(shortLinkEntity.getShortLinkCode());
+	}
+
+	@Test
+	void shouldThrowShortLinkNotFoundExceptionWhenShortLinkDoesNotExist() {
+		when(shortLinkRepository.findByShortLinkCode(shortLinkEntity.getShortLinkCode())).thenReturn(List.of());
+		assertThrows(ShortLinkNotFoundException.class, () -> apiService.getShortLink(shortLinkEntity.getShortLinkCode()));
+	}
+
+	@Test
+	void shouldDeleteShortLink() {
+		when(shortLinkRepository.existsByShortLinkCode(shortLinkEntity.getShortLinkCode())).thenReturn(true);
+		apiService.deleteShortLink(shortLinkEntity.getShortLinkCode());
+		verify(shortLinkRepository, times(1)).deleteByShortLinkCode(shortLinkEntity.getShortLinkCode());
+	}
 
 	@Test
-	void contextLoads() {
+	void shouldNotDeleteShortLink() {
+		when(shortLinkRepository.existsByShortLinkCode(shortLinkEntity.getShortLinkCode())).thenReturn(false);
+		assertThrows(ShortLinkNotFoundException.class, () -> apiService.deleteShortLink(shortLinkEntity.getShortLinkCode()));
 	}
 
-}
+	@Test
+	void shouldCreateAShortLink() {
+		String longLink = "https://news.sky.com/";
+		ShortLinkEntity savedEntity = ShortLinkEntity.builder()
+				.id(1L)
+				.longLink("https://news.sky.com/")
+				.shortLinkCode("NjxQpL")
+				.build();
+
+		when(shortLinkRepository.existsByShortLinkCode(anyString())).thenReturn(false);
+		when(shortLinkRepository.save(any(ShortLinkEntity.class))).thenReturn(savedEntity);
```

Comment:
this doesn't check what was sent to the repo in the save call.

you could use arg captors here to capture the argument that has been passed to the `save` call and verify that it is the expected one.

Reply:
https://site.mockito.org/javadoc/current/org/mockito/ArgumentCaptor.html